### PR TITLE
pythonPackages.smugline,pythonPackages.smugpy: remove

### PR DESCRIPTION
**Pull request by omasanori**
###### Motivation for this change
Upstream states that [smugline is no longer maintained](https://github.com/gingerlime/smugline#broken-no-longer-maintained) and [smugpy was abandoned](https://github.com/gingerlime/smugline#important-note).

Part of https://github.com/NixOS/nixpkgs/issues/101964

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -7187,8 +7187,6 @@ in
 
   smu = callPackage ../tools/text/smu { };
 
-  smugline = python3Packages.smugline;
```
Comment (jonringer) [inline pkgs/top-level/all-packages.nix:7190]: for top level attrs, we usually add an entry to `pkgs/top-level/aliases.nix` throwing an error stating why it was removed.

**Review by jonringer — COMMENTED** [review]
otherwise LGTM

cc @jtojnar

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (omasanori) [reply pkgs/top-level/all-packages.nix:7190]: OK, thank you for advice!

Comment (omasanori) [reply pkgs/top-level/all-packages.nix:7190]: Done.

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 102610 [1](https://github.com/Mic92/nixpkgs-review)

**Pull request merged**